src/main/java/mate/academy/spring/dao/impl/UserDaoImpl.java

```diff
@@ -20,7 +20,7 @@ public UserDaoImpl(SessionFactory factory) {
     public Optional<User> findByEmail(String email) {
         try (Session session = factory.openSession()) {
             Query<User> findByEmail = session.createQuery(
-                    "FROM User WHERE email = :email", User.class);
+                    "FROM User u join fetch u.roles WHERE u.email = :email", User.class);
```

Comment:
do follow the one style in HQL statements: capitalize or lower case for service commands
JOIN is service command too